### ISSUE TITLE
Deactivating one side of connector causes representation error

### ISSUE THREAD

**Issue by qtothec** [Pyomo/pyomo#91]
The following code creates a variable x on each of two blocks. It then links the two x's together via a Connector. Afterwards, one block and the link are deactivated. However, the code results in a KeyError in the NL writer. This is a rather urgent issue since many use cases can be found for deactivating part of a linked system of blocks.
I should note that it would be even better if I also did not need to track down and deactivate the links `m.link.deactivate()` when one side of the Connector is deactivated.
The stack trace and example can be found below:

# Small example
```
from pyomo.environ import *
from pyomo.core.base.connector import ConnectorExpander

m = ConcreteModel()
m.b1 = Block()
m.b1.x1 = Var(bounds=(0, 1))
m.b1.p1 = Connector()
m.b1.p1.add(m.b1.x1, 'X')

m.b2 = Block()
m.b2.x2 = Var()
m.b2.p2 = Connector()
m.b2.p2.add(m.b2.x2, 'X')

m.obj = Objective(expr=m.b2.x2)
m.link = Constraint(expr=m.b1.p1 == m.b2.p2)
m.b1.deactivate()
m.link.deactivate()

xfrm = ConnectorExpander()
xfrm.apply(instance=m)

ipopt = SolverFactory('ipopt')
ipopt.solve(m, tee=True)

m.display()
```

# Stack trace
```
Traceback (most recent call last):
  File "main.py", line 24, in <module>
    ipopt.solve(m, tee=True)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/opt/base/solvers.py", line 562, in solve
    self._presolve(*args, **kwds)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/solvers/plugins/solvers/ASL.py", line 188, in _presolve
    SystemCallSolver._presolve(self, *args, **kwds)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/opt/solver/shellcmd.py", line 195, in _presolve
    OptSolver._presolve(self, *args, **kwds)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/opt/base/solvers.py", line 646, in _presolve
    **kwds)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/opt/base/solvers.py", line 714, in _convert_problem
    **kwds)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/opt/base/convert.py", line 109, in convert_problem
    problem_files, symbol_map = converter.apply(*tmp, **tmpkw)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/solvers/plugins/converter/model.py", line 163, in apply
    io_options=io_options)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/core/base/PyomoModel.py", line 977, in write
    io_options)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/repn/plugins/ampl/ampl_.py", line 356, in __call__
    include_all_variable_bounds=include_all_variable_bounds)
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/repn/plugins/ampl/ampl_.py", line 870, in _print_model_NL
    list(self_varID_map[id(var)] for var in ampl_repn._linear_vars),
  File "/home/qichen/.solvers/pyomo/src/pyomo/pyomo/repn/plugins/ampl/ampl_.py", line 870, in <genexpr>
    list(self_varID_map[id(var)] for var in ampl_repn._linear_vars),
KeyError: 140220870283088
```

**Comment by gseastream** [comment]
Note this issue still persists on solvers such as `cbc` through the ampl interface. From my limited understanding, this could be fixed by implementing an approach similar to the GAMS and other writers, by iterating through the active constraints. That being said, @qtothec and I dug as far as we could into why ipopt started working and got caught up in the weird solver api so I'm not exactly sure how to implement a fix for the ampl writers. Just here to point out the issue still exists.

**Comment by blnicho** [comment]
Fixed in #583
